src/utils/debounce.ts

```diff
@@ -0,0 +1,13 @@
+/** Delay function execution until after the invocations stopped. */
+export const debounce = <T extends unknown[]>(
+  fn: (...args: T) => void,
+  period: number
+): ((...args: T) => void) => {
+  let timeout: ReturnType<typeof setTimeout> | undefined
+  return (...args: T) => {
+    clearTimeout(timeout)
+    timeout = setTimeout(() => {
+      fn(...args)
+    }, period)
+  }
```

Comment:
works me. the old throttle() implementation returned a cleanup function that allowed callers to cancel their work:

```ts
/** Limit function execution frequency. */
export function throttle<T extends unknown[]>(
  fn: (...args: T) => void,
  period: number
): (...args: T) => void {
  let timeout: ReturnType<typeof setTimeout> | undefined;
  let prev = 0;
  let nextArgs: T | undefined;
  return function (this: unknown, ...args: T): () => void {
    nextArgs = args;
    timeout ??= setTimeout(() => {
      prev = Date.now();
      timeout = undefined;
      const args = nextArgs!;
      nextArgs = undefined;
      fn.apply(this, args);
    }, Math.max(0, period - (Date.now() - prev)));
    return () => {
      if (timeout != null) clearTimeout(timeout);
      timeout = undefined;
      nextArgs = undefined;
    };
  };
}
```

ideally, we'd clean up unfinished work at least on disconnect but probably not a big deal.

Reply:
FWIW there was no cleanup with the throttle approach, but I figured it's better to have it